_posts/2020-02-10-calculate-tech-debt-using-skunk-on-github-actions.md

```diff
@@ -0,0 +1,221 @@
+
+The next thing I wanted to do is run `skunk` on a pull request in order to
+compare the _Stink Score_ between the pull request and `master`. This will help
+us answer this question:
+
+> Are we increasing or decreasing the tech debt average in our project?
+
+In order to do this, I had to tweak the call to `skunk` to use the `--branch`
+option:
+
+```yaml
+# .github/workflows/skunk.yml
+
+name: Skunk
+on: [push]
+
+jobs:
+  skunk:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Setup Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 2.6.x
+      - name: Run Skunk on Project
+        run: |
+          gem install skunk
+          CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+          if [[ "$CURRENT_BRANCH" != "master" ]]; then
+            echo "Executing within branch: $CURRENT_BRANCH"
+            skunk lib/ -b master
+          else
+            echo "Executing within master branch"
+            skunk lib/
+          fi
+```
+
+There is some `bash` _magic_ in there. The GitHub Action will do one thing if
+you are within the context of a pull request (compare action), and another thing
+if it is running a commit pushed to the `master` branch.
+
+One last thing I had to add was a step to generate [SimpleCov](https://github.com/colszowka/simplecov)'s
+resultset JSON file. Skunk is most useful when it considers code coverage data.
+[The StinkScore is a combination of code smells; complexity; and code coverage data](https://github.com/fastruby/skunk#what-is-the-stinkscore).
+
+I tweaked the steps configuration to look like this:
+
+```yaml
+steps:
+  - uses: actions/checkout@v1
+  - name: Setup Ruby
+    uses: actions/setup-ruby@v1
+    with:
+      ruby-version: 2.6.x
+  - name: Run test suite with COVERAGE=true
+    run: |
+      gem install bundler
+      bundle install --jobs 4 --retry 3
+      COVERAGE=true bundle exec rake test
+  - name: Run Skunk on Project
+    run: |
+      gem install skunk
+      CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+      if [[ "$CURRENT_BRANCH" != "master" ]]; then
+        echo "Executing within branch: $CURRENT_BRANCH"
+        skunk lib/ -b master
+      else
+        echo "Executing within master branch"
+        skunk lib/
+      fi
+```
+
+Now you can see the difference in tech debt between `master` and your pull
+request:
+
+```
+Base branch (master) average stink score: 13.42
+Feature branch ((HEAD detached at 0315f34)) average stink score: 13.42
+Score: 13.42
+```
+
+This particular example shows that there is no difference in my application's
+technical debt: [https://github.com/fastruby/skunk/pull/26](https://github.com/fastruby/skunk/pull/26).
+That's because I didn't change any Ruby code (just a YAML file).
+
+You can see the GitHub Action run over here:
+[https://github.com/fastruby/skunk/runs/437118684](https://github.com/fastruby/skunk/runs/437118684)
```

Comment:
@etagwerker fyi, it shows a syntax error there
```
running simple_cov
...............
sh: 1: Syntax error: "(" unexpected
```

Reply:
@lubc Yes, I noticed that, but I don't think it is a big issue.

Reply:
It's been reported here: https://github.com/whitesmith/rubycritic/issues/346